src/common/intl.cpp

```diff
@@ -189,6 +189,77 @@ wxString wxLanguageInfo::GetLocaleName() const
     return retval;
 }
 
+// ----------------------------------------------------------------------------
+// wxLocaleIdent
+// ----------------------------------------------------------------------------
+
+wxLocaleIdent::wxLocaleIdent(const wxString& language) : m_language(language) { }
+
+wxLocaleIdent& wxLocaleIdent::Language(const wxString& language)
+{
+    m_language = language;
+    return *this;
+}
+
+wxLocaleIdent& wxLocaleIdent::Region(const wxString& region)
+{
+    m_region = region;
+    return *this;
+}
+
+wxLocaleIdent& wxLocaleIdent::Script(const wxString& script)
+{
+    m_script = script;
+    return *this;
+}
+
+wxLocaleIdent& wxLocaleIdent::Modifier(const wxString& modifier)
+{
+    m_modifier = modifier;
+    return *this;
+}
+
+wxString wxLocaleIdent::GetName() const
+{
+    // Construct name in right format:
+    // Windows: <language>-<script>-<REGION>
+    // Unix:    <language>_<REGION>@<modifer>
+    // Mac:     <language>-<script>_<REGION>
+
+    wxString name;
+    if (!m_language.IsEmpty()) {
+        name << m_language;
+
+#if defined(__WINDOWS__) || defined(__WXOSX__)
+        if (!m_script.IsEmpty()) {
+            name << "-" << m_script;
+        }
+#endif
+
+        if (!m_region.IsEmpty()) {
+#if defined(__WINDOWS__)
+            name << "-";
+#else
+            name << "_";
+#endif
+            name << m_region;
+        }
+    }
+
+#if defined(__UNIX__) && !defined(__WXOSX__)
+    if (!m_modifer.IsEmpty()) {
+        name << "@" << m_modifer;
+    }
+#endif
+
+    return name;
+}
+
+bool wxLocaleIdent::IsDefault() const
+{
+    return m_language.IsEmpty();
```

Comment:
Very minor, but we try to use STL-compatible method names in the new code, i.e.
```suggestion
    return m_language.empty();
```